src/common/utils/api-request.ts

```diff
@@ -0,0 +1,52 @@
+import axios, { AxiosResponse } from 'axios';
+
+const apiBaseUrl: string =
+  process.env.REACT_APP_API_BASE_URL || 'http://localhost:8000';
+
+const TARGET_BASE_PATH = '/target';
+
+interface RequestParameters {
+  [key: string]:
+    | string
+    | number
+    | boolean
+    | ReadonlyArray<string>
+    | ReadonlyArray<number>
+    | ReadonlyArray<boolean>
+    | undefined
+    | null;
+}
+
+interface GetParameters {
+  path: string;
+  parameters?: RequestParameters;
+}
+
+interface ApiParameters extends RequestParameters {
+  format: 'json';
```

Comment:
Should this be string? Or can we hard code values in interfaces also? Bear with my typescript noobness :D

Reply:
I'll fix it so it uses enum as value as well